.drone.yml

```diff
@@ -7071,8 +6785,7297 @@ steps:
+    public.ecr.aws
+  - docker manifest create "public.ecr.aws/gravitational/teleport:$(echo $(cat '/go/vars/full-version/v9')
+    | sed 's/v//' | cut -d'.' -f 1,2)" --amend "public.ecr.aws/gravitational/teleport:$(echo
+    $(cat '/go/vars/full-version/v9') | sed 's/v//' | cut -d'.' -f 1,2)-amd64" --amend
+    "public.ecr.aws/gravitational/teleport:$(echo $(cat '/go/vars/full-version/v9')
+    | sed 's/v//' | cut -d'.' -f 1,2)-arm" --amend "public.ecr.aws/gravitational/teleport:$(echo
+    $(cat '/go/vars/full-version/v9') | sed 's/v//' | cut -d'.' -f 1,2)-arm64"
+  - docker manifest push "public.ecr.aws/gravitational/teleport:$(echo $(cat '/go/vars/full-version/v9')
+    | sed 's/v//' | cut -d'.' -f 1,2)"
+  - docker logout "public.ecr.aws"
+  environment:
+    AWS_ACCESS_KEY_ID:
+      from_secret: PRODUCTION_TELEPORT_DRONE_USER_ECR_KEY
+    AWS_SECRET_ACCESS_KEY:
+      from_secret: PRODUCTION_TELEPORT_DRONE_USER_ECR_SECRET
+  volumes:
+  - name: dockersock
+    path: /var/run
+  depends_on:
+  - Tag and push image "teleport:v9-amd64" to ECR - production
+  - Tag and push image "teleport:v9-arm" to ECR - production
+  - Tag and push image "teleport:v9-arm64" to ECR - production
+- name: Create manifest and push "teleport:canonical" to ECR - production
+  image: docker
+  commands:
+  - apk add --no-cache aws-cli
+  - TIMESTAMP=$(date -d @"$DRONE_BUILD_CREATED" '+%Y%m%d%H%M')
+  - aws ecr-public get-login-password --region=us-east-1 | docker login -u="AWS" --password-stdin
+    public.ecr.aws
+  - docker manifest create "public.ecr.aws/gravitational/teleport:$(echo $(cat '/go/vars/full-version/v9')
+    | sed 's/v//' | cut -d'.' -f 1,2,3)" --amend "public.ecr.aws/gravitational/teleport:$(echo
+    $(cat '/go/vars/full-version/v9') | sed 's/v//' | cut -d'.' -f 1,2,3)-amd64" --amend
+    "public.ecr.aws/gravitational/teleport:$(echo $(cat '/go/vars/full-version/v9')
+    | sed 's/v//' | cut -d'.' -f 1,2,3)-arm" --amend "public.ecr.aws/gravitational/teleport:$(echo
+    $(cat '/go/vars/full-version/v9') | sed 's/v//' | cut -d'.' -f 1,2,3)-arm64"
+  - docker manifest push "public.ecr.aws/gravitational/teleport:$(echo $(cat '/go/vars/full-version/v9')
+    | sed 's/v//' | cut -d'.' -f 1,2,3)"
+  - docker logout "public.ecr.aws"
+  environment:
+    AWS_ACCESS_KEY_ID:
+      from_secret: PRODUCTION_TELEPORT_DRONE_USER_ECR_KEY
+    AWS_SECRET_ACCESS_KEY:
+      from_secret: PRODUCTION_TELEPORT_DRONE_USER_ECR_SECRET
+  volumes:
+  - name: dockersock
+    path: /var/run
+  depends_on:
+  - Tag and push image "teleport:v9-amd64" to ECR - production
+  - Tag and push image "teleport:v9-arm" to ECR - production
+  - Tag and push image "teleport:v9-arm64" to ECR - production
+- name: Download "teleport-ent" Dockerfile and DEB artifacts from APT
+  image: ubuntu:22.04
+  commands:
+  - PACKAGE_NAME="teleport-ent"
+  - PACKAGE_VERSION="$(cat '/go/vars/full-version/v9')"
+  - apt update
```

Comment:
I feel very uneasy about this script basically being embedded in drone yaml. Can it at least be factored out in a bash script or something?

Reply:
I can do this if you like however I see several downsides:
* It adds a snowflake case where the logic behind this step is not in dronegen, rather in a shell script(s)/makefile target(s)
* We lose several compile-time/dronegen-runtime checks
* It introduces another layer of complexity
* It would move a fairly large amount of Go logic into the script (see the function that generates these commands: https://github.com/gravitational/teleport/pull/16688/files#diff-4df21a264c3b32375c8d1ed01e7de5aec5d63ca23f30721113aeb45db7d42f45R455-R549)

I'm all for a making our pipelines more CI platform agnostic but I'm not sure the tradeoffs in the case are worth it. What are your thoughts?